src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,48 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        resize();
+        stack[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
```

Comment:
how about to make new method checkEmptyStack() and use them here and in pop() as well?

Reply:
Done, made a separate private method `checkEmptyStack()`